percona/controller/pgcluster/controller.go

```diff
@@ -535,6 +539,47 @@ func (r *PGClusterReconciler) reconcileCustomExtensions(cr *v2.PerconaPGCluster)
 		extensionKeys = append(extensionKeys, key)
 	}
 
+	if cr.CompareVersion("2.6.0") >= 0 {
```

Comment:
I would put all this block in the `extensions` package (in the k8s.go file) we have and here you just call some function, like `extension.Remove` or something.